master/internal/stream/system.go

```diff
@@ -224,10 +221,12 @@ func (ps *PublisherSet) Websocket(
 	//   - fallout
 	msgs, err := ss.Startup(ctx, user, startupMsg)
 	if err != nil {
+		// XXX: won't this cause the master to crash?
 		return errors.Wrapf(err, "gathering startup messages")
 	}
 	err = writeAll(socket, msgs)
 	if err != nil {
+		// XXX: won't this cause the master to crash?
 		return errors.Wrapf(err, "writing startup messages")
```

Comment:
I don't think so?  Pretty sure this just means that this connection terminated with an error.  Am I wrong about that?  I see you changed an error to return nil above.  We should get to the bottom of that.

Reply:
I had seen some crashes when I was originally writing this, but after testing this, returning errors to `WebsocketRoute` doesn't cause a crash, not properly exiting & cleaning up will though. 

So this is a non-issue, I'll remove these comments